### ISSUE TITLE
[A11y] Advanced Search - facet link function

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#1797]
### Operating System

Windows

### Browser

Edge

### Affected page

Advanced Search

### Admin user?

- [X] Yes

### Issue Description

The facet show/hide toggles are all coded as links, but aren't really links.  They're even styled to look a bit like buttons, and not at all like links, which is confusing.

They seem to function as a disclosure more than any other widget, but aren't marked up or coded as such.

### Suggested Fix

If we want these to be disclosures we should code them as such: https://www.w3.org/TR/wai-aria-practices/#disclosure

They could continue to use the `<a>` element so long as they are coded with all the proper ARIA and JavaScript, though buttons would be a bit easier since they natively support spacebar for activation, which is required for the disclosure pattern.

**Comment by jechols** [comment]
Update: I believe part of the problem is the div being the clickable entity, but not being an actual control.  Looking at the code I'm seeing that the surrounding divs actually have ARIA attributes on them.  But those aren't seen by screen readers since the divs aren't real controls and never get focus.

**Comment by CGillen** [comment]
@jechols Would this be closer to an accordion than a disclosure?
https://www.w3.org/TR/wai-aria-practices-1.1/#accordion
In which case, converting to a button would give us the basic requirements, yeah?
Enter & Space collapses/expands. We are not requiring only one facet be expanded at a time.
Tab and Shift+Tab moves along the tab path of links.
The others are optional

**Comment by jechols** [comment]
@CGillen I think you're right.  I thought that the accordion required only one to be visible at a time, but it's clear that's optional and an implementation decision.  An accordion definitely seems like the right fit.

My mistake, sorry about that :(

**Comment by jechols** [comment]
Same failure as other accordions:

>  The accordion header button element has aria-controls set to the ID of the element containing the accordion panel content.
